source/common/network/filter_manager_impl.cc

```diff
@@ -81,10 +82,13 @@ FilterStatus FilterManagerImpl::onWrite(ActiveWriteFilter* filter,
   }
 
   for (; entry != downstream_filters_.end(); entry++) {
-    StreamBuffer write_buffer = buffer_source.getWriteBuffer();
-    FilterStatus status = (*entry)->filter_->onWrite(write_buffer.buffer, write_buffer.end_stream);
-    if (status == FilterStatus::StopIteration) {
-      return status;
+    if (connection_.state() == Connection::State::Open) {
```

Comment:
You can pull this outside the loop.

Reply:
Again, prefer a check only after a filter runs.